### PR TITLE
[terra-alert] Add Unsatisfied and Unverified Types

### LINKED ISSUE #2844: Alert view has no bottom border is causing visual defect
# Bug Report

## Description

When we have header or alert view stacked under an alert view, there's no separator between them. 
## Steps to Reproduce

Create two stacked alert view.

or

 Add a header view under an alert view

## Additional Context / Screenshots

Two alerts stacking vertically:
conditions_two_alerts

Alert with header (See chronic no recorded and allergies part):
screenshot-1

## Expected Behavior

There should be a bottom border for alert view in above scenario.
## Possible Solution

This issue may occur with a variety of scenarios, as long as the component beneath alert view has same color and doesn't have a top border. We can add the alert to a list with bottom border only and no padding, but it seems hacky and brings in extra dependencies.

  The bottom border should be an option of alert view to avoid this issue.

## Environment

* Component Name and Version: terra-alert 4.20
* Browser Name and Version: Any
* Node/npm Version: [e.g. Node 8/npm 5] 
* Webpack Version:
* Operating System and version (desktop or mobile):

### LINKED ISSUE #2928: [terra-alert] Add Unsatisfied and Unverified Types
# Feature Request

## Description
Add unsatisfied and unverified alert types. 

### unsatisfied
Screen Shot 2020-04-15 at 11 06 29 AM

### unverified
Screen Shot 2020-04-15 at 11 06 38 AM

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary

- Add Unsatisfied and Unverified Types.
    - `unsatisfied` - recommended to provide the user that an unsatisfied or incomplete action needs to be addressed
    - `unverified` - recommended to provide feedback that information from outside systems have been included but have yet to be verified by the user
- Updated to use theme-context to apply orion-fusion-theme and lowlight theme
- Updated the doc and test examples to use hooks
- Added bottom margin to fix visual defect when multiple alerts are stacked

Closes #2928 Closes #2844 

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Testing

- Added jest & wdio tests for new types
- Update tests to stack alerts for UX validation
- Update/renamed misleading existing tests
- existing tests

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-alert/translations/ar.json [hunk 1]
```diff
@@ -1,9 +0,0 @@
-{
-  "Terra.alert.dismiss": "تجاهل",
```
Comment (emilyrohrbough) [inline packages/terra-alert/translations/ar.json:2]: Talked offline to no request new translations for these types. Should I keep the existing translations even with the missing keys?

File: packages/terra-alert/translations/fi-FI.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
```
Comment (emilyrohrbough) [inline packages/terra-alert/translations/fi-FI.json:1]: We don't support finish so didn't request new translations for these types. Should I keep the existing translations even with the missing keys?

File: packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss [hunk 3]
```diff
@@ -1,41 +1,3 @@
...
-  --terra-alert-color: #1c1f21;
-  --terra-alert-min-height: 3.41667rem;
-  --terra-alert-icon-font-size: 1.3333em;
-  --terra-alert-icon-margin-right: 0.475rem;
-  --terra-alert-icon-margin-left: 0;
-  --terra-alert-icon-margin-top: 0;
-  --terra-alert-body-margin-left: 0.5rem;
-  --terra-alert-body-margin-right: 0.5rem;
-  --terra-alert-body-margin-top: 0.75rem;
   --terra-alert-narrow-margin-right: 0.5rem;
```
Comment (emilyrohrbough) [inline packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss:2]: This is the only theme var that isn't used. Not sure if it's a miss or if we need to keep for backwards compatibility?

File: packages/terra-alert/src/Alert.module.scss [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+@import './orion-fusion-theme/Alert.module';
+
 :local {
   /* stylelint-disable  property-no-unknown */
   .alert-base {
     align-items: stretch;
     color: var(--terra-alert-color);
     display: flex;
     justify-content: space-between;
+    margin: 0 0 1px;
```
Comment (emilyrohrbough) [inline packages/terra-alert/src/Alert.module.scss:12]: fixes #2844

File: packages/terra-alert/src/Alert.module.scss [hunk 5]
```diff
@@ -137,6 +152,14 @@
...
   }
 
+  .unsatisfied-icon {
+    color: var(--terra-alert-unsatisfied-icon-color, #fa0000);
+  }
+
+  .unverified-icon {
+    color: var(--terra-alert-unverified-icon-color, #78288c);
+  }
+
```
Comment (emilyrohrbough) [inline packages/terra-alert/src/Alert.module.scss:162]: These icons are themeable icons & needed a color applied to be styled correctly

File: packages/terra-alert/src/Alert.jsx [hunk 6]
```diff
@@ -25,7 +32,8 @@ const AlertTypes = {
 
 const propTypes = {
   /**
-   * An action element to be added to the action section of the alert.
+   * An action element to be added to the action section of the alert to give the user an easy ways
```
Comment (benbcai) [inline packages/terra-alert/src/Alert.jsx:35]:
```suggestion
   * An action element to be added to the action section of the alert to give the user an easy way
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 7]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
 
 # Terra Alert
 
-The Terra Alert component is a notification banner that can be rendered in your application when there is information that you want to bring to the user's attention.  The Alert component supports a number of built-in notification types that render with pre-defined colors and icons that help the user understand the severity and meaning of the notification. A custom notification type is also supported that allows your application to customize an alert that may not fit into the pre-defined types.
+The Terra Alert component is a notification banner that can be rendered in your application when there is information that you want to bring to the user's attention.
 
+The Alert component supports a number of built-in notification types that render with pre-defined colors and icons that help the user understand the severity and meaning of the notification. The support alert types and recommended uses are:
```
Comment (tbiethman) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:25]:
```suggestion
The Alert component supports a number of built-in notification types that render with pre-defined colors and icons that help the user understand the severity and meaning of the notification. The supported alert types and recommended uses are:
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 8]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
...
+
+- `alert` - recommended to display a critical notifications for current emergencies or situations of high severity that need to be addressed promptly
```
Comment (tbiethman) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:27]:
```suggestion
- `alert` - recommended to display a critical notification for current emergencies or situations of high severity that need to be addressed promptly
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 9]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
...
+- `error` - recommended to provide feedback of a negative system status such as failures, inabilities, or limitations
+- `warning` - recommended to show moderately urgent notifications that are cautionary in nature and forecast potential risk
+- `unsatisfied` - recommended to provide the user that an unsatisfied or incomplete action needs to be addressed
```
Comment (tbiethman) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:30]:
```suggestion
- `unsatisfied` - recommended to show the user that an unsatisfied or incomplete action needs to be addressed
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 10]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
...
+- `unverified` - recommended to provide feedback that information from outside systems have been included but have yet to be verified by the user
```
Comment (tbiethman) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:31]:
```suggestion
- `unverified` - recommended to provide feedback that information from outside systems has been included but has yet to be verified by the user
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 11]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
...
+- `advisory` - recommended to provide the user with guidance and/or advice for completing an action
+- `information` - recommended to provide general awareness and information that can possibly lead to an action for something that but has no associated severity.
```
Comment (tbiethman) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:33]:
```suggestion
- `information` - recommended to provide general awareness and information that can possibly lead to an action for something that has no associated severity.
```

File: packages/terra-alert/translations/fi-FI.json (hunk 2 above)
Comment (tbiethman) [reply packages/terra-alert/translations/fi-FI.json:1]: I vote to remove them if we don't support the language.

File: packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss (hunk 3 above)
Comment (tbiethman) [reply packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss:2]: I looked in the history and ... cannot find usage of it anywhere, past or present. 😕

Comment (emilyrohrbough) [reply packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss:2]: Shall we remove it?? I'm okay with it if everyone else is!

Comment (mjhenkes) [reply packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss:2]: Are we 100% sure? These aren't used in the consumer theme, clinical theme, lowlight or fusion themes?

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 12]
```diff
@@ -18,8 +20,20 @@ import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
 
 # Terra Alert
 
-The Terra Alert component is a notification banner that can be rendered in your application when there is information that you want to bring to the user's attention.  The Alert component supports a number of built-in notification types that render with pre-defined colors and icons that help the user understand the severity and meaning of the notification. A custom notification type is also supported that allows your application to customize an alert that may not fit into the pre-defined types.
+The Terra Alert component is a notification banner that can be rendered in your application when there is information that you want to bring to the user's attention.
 
+The Alert component supports a number of built-in notification types that render with pre-defined colors and icons that help the user understand the severity and meaning of the notification. The supported alert types and recommended uses are:
+
+- `alert` - recommended to display a critical notification for current emergencies or situations of high severity that need to be addressed promptly
```
Comment (yuderekyu) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx:27]: we mention these are recommendations above; should we drop "recommended" from these descriptions then?
```suggestion
- `alert` - displays a critical notification for current emergencies or situations of high severity that need to be addressed promptly
```

File: packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss (hunk 3 above)
Comment (emilyrohrbough) [reply packages/terra-alert/themes/orion-fusion-theme/orion-fusion-theme.scss:2]: 🤷‍♀️

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx (hunk 12 above)
Comment (yuderekyu) [reply packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx]: ah, for consistency, if we make this change it should be carried through to the other descriptions

File: packages/terra-alert/src/Alert.jsx [hunk 13]
```diff
@@ -107,28 +116,29 @@ const Alert = ({
...
-  const attributes = { ...customProps };
-  const alertClassNames = cx([
-    'alert-base',
-    type,
-    { narrow: isNarrow },
-    { wide: !isNarrow },
-    attributes.className,
-    { [`${customColorClass}`]: type === AlertTypes.CUSTOM },
+  const alertClassNames = classNames([
+    cx([
```
Comment (StephenEsser) [inline packages/terra-alert/src/Alert.jsx:124]: The array brackets for classnames is optional. This syntax is fine, I just wanted to throw that out there for future reference. Classnames and cx will both spread arguments as an array internally. 

```js
cx(['one', 'two', 'three']) === cx('one', 'two', 'three')
```

Comment (emilyrohrbough) [reply packages/terra-alert/src/Alert.jsx]: Good point. I'll simplify now

File: packages/terra-alert/src/clinical-lowlight-theme/Alert.module.scss [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+    --terra-alert-body-margin-left: 0.64285rem;
+    --terra-alert-body-margin-right: 0.7143rem;
+    --terra-alert-body-margin-top: 0.7143rem;
+
+    --terra-alert-std-margin-bottom: 0.7143rem;
+    --terra-alert-std-margin-right: 0.6667rem;
+
+    --terra-alert-section-padding-bottom: 0.142857rem;
+    --terra-alert-section-padding-top: 0.142857rem;
```
Comment (jeremyfuksa) [inline packages/terra-alert/src/clinical-lowlight-theme/Alert.module.scss:19]: Setting this to `0.214285714rem` improves alignment with icons.

Comment (emilyrohrbough) [reply packages/terra-alert/src/clinical-lowlight-theme/Alert.module.scss]: Updated:
- default-theme - https://github.com/cerner/terra-core/pull/2941/commits/bc9d52db9887bcdb9ce86d86a60fa3c13ce0ce25#diff-3b62f1ed274440a76fc2e27af8d4700bR62
- lowlight-theme - https://github.com/cerner/terra-core/pull/2941/commits/bc9d52db9887bcdb9ce86d86a60fa3c13ce0ce25#diff-15ad366066734dcfa4399a813916cc0cR15

**Comment by PaulSolDev** [comment]
Functional review complete. No issues identified.

**Pull request merged**